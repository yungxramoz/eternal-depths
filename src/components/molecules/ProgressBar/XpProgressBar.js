import React from 'react'
import RpgProgressBar from '../../atoms/RpgProgressBar/RpgProgressBar'
import './ProgressBar.css'

const XpProgressBar = ({ currentXp, maxXp }) => {
  return (
    <div className="progress-container">
      <p className='hint-text'>
        XP {currentXp}/{maxXp}
      </p>
      <RpgProgressBar max={maxXp} current={currentXp} type="xp" />
    </div>
  )
}

export default XpProgressBar
