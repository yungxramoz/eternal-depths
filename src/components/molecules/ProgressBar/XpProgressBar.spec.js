import { render, screen } from '@testing-library/react'
import React from 'react'
import XpProgressBar from './XpProgressBar'

describe('XpProgressBar', () => {
  it('renders correctly', () => {
    render(<XpProgressBar maxXp={100} currentXp={50} />)
    expect(screen).toMatchSnapshot('XpProgressBar')
  })

  it('displays correct XP values', () => {
    render(<XpProgressBar maxXp={100} currentXp={50} />)
    expect(screen.getByText('XP 50/100')).toBeInTheDocument()
  })
})
