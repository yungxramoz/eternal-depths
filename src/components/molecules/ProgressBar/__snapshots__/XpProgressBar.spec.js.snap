// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XpProgressBar renders correctly: XpProgressBar 1`] = `
<body>
  <div>
    <div
      class="progress-container"
    >
      <p
        class="hint-text"
      >
        XP 
        50
        /
        100
      </p>
      <div
        class="rpgui-progress"
        data-rpguitype="progress"
      >
        <div
          class=" rpgui-progress-track"
        >
          <div
            class="rpgui-progress-fill purple progress-animation"
            style="left: 0px; width: 50%;"
          />
        </div>
        <div
          class=" rpgui-progress-left-edge"
        />
        <div
          class=" rpgui-progress-right-edge"
        />
      </div>
    </div>
  </div>
</body>
`;
